test/index.js

```diff
@@ -1 +1,1159 @@
+          cwd: cwd,
+        },
+        expected: {
+          path: path.resolve(cwd, 'test/fixtures/fined'),
+          extension: '',
+        },
+      },
+      {
+        name: 'When \'name\' is an empty string : ' +
+              'ignored and use only \'path\'',
+        pathObj: {
+          name: '',
+        },
+        defaultObj: {
+          path: 'test/fixtures/fined',
+          extensions: [''],
+          cwd: cwd,
+        },
+        expected: {
+          path: path.resolve(cwd, 'test/fixtures/fined'),
+          extension: '',
+        },
+      },
+      {
+        name: 'When \'name\' is not a string : ignored and use only \'path\'',
+        pathObj: {
+          name: 123,
+        },
+        defaultObj: {
+          path: 'test/fixtures/fined',
+          extensions: [''],
+          cwd: cwd,
+        },
+        expected: {
+          path: path.resolve(cwd, 'test/fixtures/fined'),
+          extension: '',
+        },
+      },
+      {
+        name: 'When \'name\' is a String object',
+        pathObj: {
+          name: new String('app'),
+        },
+        defaultObj: {
+          path: 'test/fixtures/fined',
+          extensions: ['.js'],
+          cwd: cwd,
+        },
+        expected: {
+          path: path.resolve(cwd, 'test/fixtures/fined', 'app.js'),
+          extension: '.js',
+        },
+      },
+      {
+        name: 'When \'name\' is a String object (2)',
+        pathObj: {
+          name: new String('package'),
+        },
+        defaultObj: {
+          path: '',
+          extensions: ['.json'],
+          cwd: cwd,
+        },
+        expected: {
+          path: path.resolve(cwd, 'package.json'),
+          extension: '.json',
+        },
+      },
+      {
+        name: 'When \'name\' indicates a directory',
+        pathObj: {
+          name: 'fined',
+        },
+        defaultObj: {
+          path: 'test/fixtures',
+          extensions: [''],
+          cwd: cwd,
+        },
+        expected: {
+          path: path.resolve(cwd, 'test/fixtures', 'fined'),
+          extension: '',
+        },
+      },
+      {
+        name: 'When \'name\' is an absolute path and \'path\' is empty',
+        pathObj: {
+          name: path.resolve(cwd, 'test/fixtures/fined/app'),
+        },
+        defaultObj: {
+          path: '',
+          extensions: ['.js'],
+          cwd: cwd,
+        },
+        expected: {
+          path: path.resolve(cwd, 'test/fixtures/fined/app.js'),
+          extension: '.js',
+        },
+      },
+      {
+        name: 'When \'name\' is an absolute path and \'path\' is not empty',
+        pathObj: {
+          name: path.resolve(cwd, 'test/fixtures/fined/app'),
+          path: 'test/fixtures/fined',
+        },
+        defaultObj: {
+          extensions: ['.js'],
+          cwd: cwd,
+        },
+        expected: null,
+      },
+      {
+        name: 'When \'name\' is an absolute path and \'path\' is empty (2)',
```

Comment:
I'm sorry that I mistook to set a test data of this test case. I modified this test data.